projects/packages/stats-admin/src/class-rest-controller.php

```diff
@@ -41,33 +58,41 @@ public function register_rest_routes() {
 				'permission_callback' => array( $this, 'check_user_privileges_callback' ),
 			)
 		);
+
+		// Stats for a resource type.
 		register_rest_route(
 			static::$namespace,
 			sprintf( '/sites/%d/stats/(?P<resource>[\-\w]+)', Jetpack_Options::get_option( 'id' ) ),
 			array(
 				'methods'             => WP_REST_Server::READABLE,
-				'callback'            => array( $this, 'get_stats_resource_from_wpcom' ),
+				'callback'            => array( $this, 'get_stats_resource' ),
 				'permission_callback' => array( $this, 'check_user_privileges_callback' ),
 			)
 		);
+
+		// Single post info.
 		register_rest_route(
```

Comment:
Have you consider completely removing this endpoint and use the WP core endpoint instead in frontend to fetch single post info?